include/appbase/application.hpp

```diff
@@ -252,16 +261,23 @@ namespace appbase {
 
       private:
          application(); ///< private because application is a singleton that should be accessed via instance()
-         map<string, std::unique_ptr<abstract_plugin>> plugins; ///< all registered plugins
-         vector<abstract_plugin*>                  initialized_plugins; ///< stored in the order they were started running
-         vector<abstract_plugin*>                  running_plugins; ///< stored in the order they were started running
+
+         // members are ordered taking into account that the last one is destructed first
+         std::shared_ptr<boost::asio::io_service>  io_serv;
```

Comment:
No reason for this to be a `shared_ptr`. It can just be a member of application.

Reply:
done